src/SFA.DAS.Authorization.EmployerFeatures/Models/EmployerFeatureToggle.cs

```diff
@@ -1,26 +1,26 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Text.RegularExpressions;
 using SFA.DAS.Authorization.Features.Models;
 
 namespace SFA.DAS.Authorization.EmployerFeatures.Models
 {
     public class EmployerFeatureToggle : FeatureToggle
     {
-        public List<EmployerFeatureToggleWhitelistItem> Whitelist { get; set; }
-        public bool IsWhitelistEnabled => Whitelist != null && Whitelist.Count > 0;
+        public List<long> AccountWhitelist { get; set; } = new List<long>();
+        public List<string> EmailWhitelist { get; set; } = new List<string>();
+        public bool IsWhitelistEnabled =>
+            (AccountWhitelist != null && AccountWhitelist.Count > 0)
+            ||
+            (EmailWhitelist != null && EmailWhitelist.Count > 0);
 
         public bool IsUserWhitelisted(long? accountId, string userEmail)
         {
-            return accountId.HasValue
-                ? Whitelist.Any(w => (w.AccountId == accountId || w.AccountId == null) && IsUserWhitelisted(userEmail))
-                : IsUserWhitelisted(userEmail);
-        }
+            return (accountId.HasValue && AccountWhitelist.Contains(accountId.Value))
+                ||
+                (!string.IsNullOrWhiteSpace(userEmail) && EmailWhitelist.Any(email => Regex.IsMatch(userEmail, email, RegexOptions.IgnoreCase)));
```

Comment:
As a matter of interest is Regex.IsMatch() quicker that than the c# Equals() & OrdinalIgnoreCase. Also possible to get a partial match (ie will A@AAA.COM match AAA@AAA.COM)

or could you use EmailWhiteList.Contains(userEmail, StringComparer.InvariantCultureIgnoreCase)

I'm not sure which option is optimal

Reply:
So some improvements could be made here. The switch to Regex wasn't performance based. The Das Auth implementation in MA allowed for wildcarding, so some of the config is set as `"EmailWhitelist":[".*@eoi.com","nlapptest2@mailinator.com"]}` 

Where a .Contains doesn't satisfy that criteria of matching any eoi.com domain email address. If there are concerns over speed, we could check a .Contains first in an attempt to short circuit and fall back on the Regex. Currently in live there are only about 26 email addresses in the config, and no wildcarding. So happy to amend this if we think there is an issue.

A partial match I hadn't considered, I'll write a unit test for that and maybe see if I can change it to be stricter where an email address is concerned.

Reply:
I've updated with a second commit to hopefully address the partial match and to also leverage the string comparer and equals for exact matches.